api/apps/api/src/modules/projects/projects.controller.ts

```diff
@@ -523,4 +523,22 @@ export class ProjectsController {
     }
     return { id: exportIdOrError.right };
   }
+
+  @ImplementsAcl()
+  @Get(':projectId/editing-locks')
+  @ApiOperation({ summary: 'Get all locks by project' })
+  async findLocksForProject(
+    @Param('projectId', ParseUUIDPipe) projectId: string,
+    @Req() req: RequestWithAuthenticatedUser,
+  ): Promise<ScenarioLockResultPlural> {
+    const result = await this.projectsService.findAllLocks(
+      projectId,
+      req.user.id,
+    );
+    if (isLeft(result)) {
+      throw new ForbiddenException();
```

Comment:
i'm not sure this will cover (and keep covering, in case we add/change stuff upstream in the call stack) all the possible errors we may want to expose to API consumers, but on the other hand I'd not want to go down the bulldozer route of exhaustive checks here, given that this is a fairly unexciting corner of the API. Just an invitation to think whether we may need to handle other kinds of HTTP responses here.

Reply:
tl;dr I think we're covering all we need and all we can expect right now - if we add more failure cases upstream in the future, we can think about whether we need different HTTP error statuses